src/Network/Wai/Middleware/Auth.hs

```diff
@@ -315,6 +316,13 @@ userKey = unsafePerformIO Vault.newKey
 getAuthUser :: Request -> Maybe AuthUser
 getAuthUser = Vault.lookup userKey . vault
 
+-- | Get the username for the current user from the given Vault.
+--
+-- Use this with Servant.Api.Vault.
+--
+-- @since 0.2.4.2
+getAuthUserVault :: Vault.Vault -> Maybe AuthUser
```

Comment:
Minor thought: I think I'd prefer `getAuthUserFromVault`.

Reply:
I prefer that too